src/main/java/usecases/suggest_title/StInteractor.java

```diff
@@ -0,0 +1,79 @@
+package usecases.suggest_title;
+import entities.SuggestedTitleChecker;
+import usecases.Response;
+
+import java.util.Arrays;
+
+/**
+ * The interactor for this use case. Contains a thread that carries out the processes involved in suggesting a story\
+ * title.
+ */
+public class StInteractor {
+    StOutputBoundary pres;
+    StGateway repo;
+    SuggestedTitleChecker titleChecker;
+
+    /**
+     * This nested class is a thread for the processes involved in suggesting a title for the story. The tasks done
+     * by the run() method in this thread are:
+     * 1. Processing the title input by the user. (trimmed and multiple whitespaces replaced with single whitespace)
+     * 2. Constructing the input data object corresponding to getting all previously suggested titles from the repo.
+     * 3. Returning a failure response if the title suggested by the user is not a valid title or has
+     *    already been suggested.
+     * 4. If no failure response is returned in the previous step, constructing the input data object corresponding to
+     *    adding the suggested title to the database.
+     * 5. Running the method to add the suggested title, and, depending on the success or failure of this process,
+     *    returning an appropriate response.
+     * 6. Passing the output data to the viewModel to update the view according to the success or failure of the
+     *    request to change title.
+     */
+    public class StThread implements Runnable{
+        StInputData data;
+        StOutputBoundary boundary;
+
+        /**
+         * The implementation of the run() method for this Thread. Performs all the tasks as specified in the
+         * description for StThread.
+         */
+        public void run(){
```

Comment:
Make this method cleaner. Add more blank lines in the code and add comments. It looks highly unreadable

Reply:
i did the necessary changes, added a lot more documentation to explain what this method does. I also put necessary comments to explain the large if ... else blocks in the code. I was thinking of breaking it up into private helper methods, but that might make the class overpopulated with helper methods, and all of the processes here are necessary to the thread anyway. Hence, I havent changed anything except for documentation.